components/nimbus/ios/Nimbus/Nimbus.swift

```diff
@@ -0,0 +1,229 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+public class Nimbus {
+    private let nimbusClient: NimbusClientProtocol
+
+    private let errorReporter: NimbusErrorReporter
+
+    private let dbQueue = DispatchQueue(label: "com.mozilla.nimbus.ios-db", qos: .userInitiated)
+    private let fetchQueue = DispatchQueue(label: "com.mozilla.nimbus.ios-network", qos: .background)
+
+    internal init(nimbusClient: NimbusClientProtocol,
+                  errorReporter: @escaping NimbusErrorReporter) {
+        self.errorReporter = errorReporter
+        self.nimbusClient = nimbusClient
+    }
+}
+
+private extension Nimbus {
+    func catchAll<T>(_ thunk: () throws -> T?) -> T? {
+        do {
+            return try thunk()
+        } catch {
+            errorReporter(error)
+            return nil
+        }
+    }
+
+    func catchAll(_ queue: DispatchQueue, thunk: @escaping () throws -> Void) {
+        queue.async {
+            self.catchAll(thunk)
+        }
+    }
+}
+
+// Glean integration
+private extension Nimbus {
+    func recordExposure(experimentId _: String) {
+        // TODO: https://jira.mozilla.com/browse/SDK-209
+    }
+
+    func postEnrolmentCalculation(_ events: [EnrollmentChangeEvent]?) {
```

Comment:
I was about to complain about 'enrolment' vs 'enrollment' here but then I noticed that nimbus-sdk already has a heady mix of both spellings :-/

I think all of the names visible in our API layer on Android use 'enrollment', FWIW.

Reply:
This, I suspect is the next front in the ongoing  `en-US` v `en-ROW` _guerre des langues_.

"Enrolment" will now join "Colour" and "Centre" in one of the most poignant `grep`-based ceremonies in Computer Science.